x-pack/legacy/plugins/maps/public/layers/heatmap_layer.js

```diff
@@ -28,12 +27,14 @@ export class HeatmapLayer extends VectorLayer {
     if (!style) {
       const defaultStyle = HeatmapStyle.createDescriptor();
       this._style = new HeatmapStyle(defaultStyle);
+    } else {
+      this._style = style;
```

Comment:
This is not needed. Default layer constructor already sets `this._style = style`

Reply:
Removed style from the layer-constructor parameters